### PR TITLE
Rewrite the missing links checks

### PR DESCRIPTION
**Pull request by benhyland**
Motivation:

We have been looking at the output from the missing links checks and found that they contain lots of 'missing' links which are actually reported because the source item for the links is actually missing from Rummager or Publishing API.

We are really more interested in items which exist in both Rummager and Publishing API for which the links are not the same.

When we originally wrote these checks, we were not explicitly importing the items in separate tables, and we didn't have as reliable a way to map Rummager base_paths to Publishing API content_ids.

Our lookup-by-base-path endpoint should now be capable of matching up all the items we have.
So our check could now use that lookup table to exclude content which doesn't exist in both stores from consideration.

Approach:

Replace the two existing checks with a single check that produces two reports.

The new check only cares about links which are on items for which
we can establish the mapping of base_path to content_id.

It does this filtering in code - my first attempt at doing the
4 joins necessary for the SQL equivalent was way too slow
(I might have missed an easier way, though).

The size of the new report for links missing from Rummager is massively reduced.
The size of the new report for links missing from Publishing API is almost unchanged.

### CODE REVIEW

**Comment by tijmenb** [comment]
Can you expand on the "why?" for the PR? What's this change going to do?

File: bin/run_automated_checks [hunk 1]
```diff
@@ -3,4 +3,4 @@
 set -e
 
 # all checks get run if none are specified
-bundle exec bin/run_checks.rb
+bundle exec bin/run_checks.rb $@
```
Comment (jackscotti) [inline bin/run_automated_checks:6]: what is `$@`?

Comment (benhyland) [inline bin/run_automated_checks:6]: See http://ss64.com/bash/syntax-parameters.html
Basically it expands to 'all arguments given in the current scope'
But see also http://ss64.com/bash/shift.html

**Comment by Davidslv** [comment]
This looks good to me, if you would like to add the spec we talked about on Friday I will then merge it.

**Comment by benhyland** [comment]
@Davidslv added a spec as discussed

**Comment by Davidslv** [comment]
👍

**Pull request merged**